Added backup url support

marcin-adamczewski opened the pull request:
Hi @pedroSG94 
I added support for the backup URL. We started using this in our app.
The behavior is that when connection fails and retry is triggered then the backup URL is being used, for the next retry if the backup URL fails, the original URL is being used, and so on.

I moved the assignment of `SrsFlvMuxer.url` to the `start()` method so it's not overridden by the `backupUrl` in the `connect()` method.
I implemented it only for RTMP. If the implementation is fine I can try to do this for RTSP as well, however, I'm not sure if I can easily test it.

pedroSG94 commented:
I like the idea but no the implementation. I think that add the url in the reconnect method is easier.
After that you can handlewhere do you wan to connect in each retry.
Example code:
```java
//replace old url to the new one
if (rtmpCamera1.reTry(5000, reason, newendpoint)) {
}
//no endpoint setted using old one
if (rtmpCamera1.reTry(5000, reason)) {
}
```
SrsFlvMuxer modification is minimal:
```java
public void reConnect(final long delay) {
    reConnect(delay, null);  
  }
  
public void reConnect(final long delay, String newEndpoint) {
    reTries--;
    url = newEndpoint == null ? url : newEndpoint;
    stop(null);
    runnable = new Runnable() {
      @Override
      public void run() {
        start(url, true);
      }
    };
    handler.postDelayed(runnable, delay);
}
```

After that, you only need to expose overrided methods to rtmp classes.
The adventage of this is that the same modification is valid for rtsp library:
https://github.com/pedroSG94/rtmp-rtsp-stream-client-java/blob/master/rtsp/src/main/java/com/pedro/rtsp/rtsp/RtspClient.kt#L346

marcin-adamczewski commented:
Good point. I will modify this according to your suggestions

marcin-adamczewski commented:
@pedroSG94 I changed the code according to your suggestions. Note that I moved deprecated reTry() method logic to the new private reTry() method, otherwise, I would have to add `String backupUrl` to this deprecated one which would cause users that migrates to the new version to unconsciously pass `reason` as a backupUrl.

I also added `onConnectionStarted(rmptUrl: String)` callback - this might be useful for analytics to know which URL you're connecting to. I added it in the second commit which is not visible for some reason in the PR :/ I hope it's gonna be visible soon.

I tested only rtmp using RtmpCamera1 and RtmpCamera2.

pedroSG94 commented:
I merged this after few modifications like rename callback, move it and remove deprecated retry method:
https://github.com/pedroSG94/rtmp-rtsp-stream-client-java/commit/b75c563811b6545f1d6963d5612cdc395394857b
